packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx

```diff
@@ -121,52 +121,23 @@ export const useAuthenticator = (selector?: Selector) => {
   };
 
   /**
-   * For `useSelector`'s selector argument, we just return back the `state`.
-   * The reason is that whenever you select a specific value of the state, the
-   * hook will return *only* that selected value instead of the whole `state`.
+   * For `useSelector`'s selector argument, we transform `state` into
+   * public facade values using `getFacade`.
    *
-   * To provide a consistent set of facade, we let the `selector` trivially return
-   * itself and let comparator decide when to re-render.
+   * This is to hide the internal xstate implementation details to customers.
    */
-  const xstateSelector = (state: AuthMachineState) => state;
-
-  /**
-   * Holds a snapshot copy of last previous facade values. Will be used
-   * on state changes to see if any of facade values have changed.
-   */
-  const prevFacadeRef = React.useRef<ReturnType<typeof getFacade>>();
+  const xstateSelector = (state: AuthMachineState) => getFacade(state);
 
   /**
    * comparator decides whether or not the new authState should trigger a
    * re-render. Does a deep equality check.
    */
   const comparator = (
-    /**
-     * We do not use `_prevState`, because it holds a *reference* to actor
-     * object, of which value could easily mutate between compare calls.
-     *
-     * Instead, we'll use prevFacadeRef for comparison.
-     */
```

Comment:
This is no longer needed for two reasons:

1. `ref` comparison is done internally from React 18's `useSyncExternalStore` hook:

https://github.com/facebook/react/blob/ce13860281f833de8a3296b7a3dad9caced102e9/packages/use-sync-external-store/src/useSyncExternalStoreShimClient.js#L131-L140

2. With `selector` now returning a flat facade values instead of xstate `state` references, we're free of the reference comparison problem mentioned in this comment.